public/js/monitor.js

```diff
@@ -51,15 +51,68 @@ function doOauth(el) {
     return;
   }
 
-  // Preserve entire control function
+  if (!emailWatch) {
+    // Preserve entire control function
+    if (sessionStorage && sessionStorage.length > 0) {
+      const lastScannedEmail = sessionStorage.getItem("lastScannedEmail");
+      if (lastScannedEmail) {
+        url.searchParams.append("email", lastScannedEmail);
+      }
+    }
+    window.location.assign(url);
+    return;
+  }
+
+  let email = false;
+
+  if (document.querySelector("#scan-user-email input[type=email]")) {
+    email = document.querySelector("#scan-user-email input[type=email]").value;
+
+    if (!isValidEmail(email)) {
+      email = false;
+    }
+  }
+
+  const scannedEmailId = document.querySelector("#scan-user-email input[name=scannedEmailId]");
+
   if (sessionStorage && sessionStorage.length > 0) {
+
     const lastScannedEmail = sessionStorage.getItem("lastScannedEmail");
-    if (lastScannedEmail) {
-      url.searchParams.append("email", lastScannedEmail);
+
+
+    if (email && lastScannedEmail) {
+      switch (email) {
+        case lastScannedEmail:
+          // The last saved email address and the current entry match, so route it to FxA
+          email = lastScannedEmail;
+          break;
+        case !lastScannedEmail:
+          // The last saved email address and the current entry DIFFER, so create
+          // a new entry, launch a new FxA login session with new email prefilled.
+          sessionStorage.removeItem("lastScannedEmail");
+          sessionStorage.setItem("lastScannedEmail", email);
+          scannedEmailId.value = sessionStorage.length;
+          break;
+      }
+    } else {
```

Comment:
`else` sniff. any way to make this a helper function with an early `return` statement?